### PR TITLE
refactor: Migrate FaviconViewModel to React

### PR DESCRIPTION
**Pull request by przemvs**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [ ] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

_Briefly describe the issue you have solved or implemented with this pull request. If the PR contains multiple issues, use a bullet list._

### Causes (Optional)

_Briefly describe the causes behind the issues. This could be helpful to understand the adopted solutions behind some nasty bugs or complex issues._

### Solutions

_Briefly describe the solutions you have implemented for the issues explained above._

### Dependencies (Optional)

_If there are some other pull requests related to this one (e.g. new releases of frameworks), specify them here._

Needs releases with:

- [ ] GitHub link to other pull request

#### Test Coverage (Optional)

- [ ] I have added automated test to this contribution

#### How to Test

_Briefly describe how this change was tested and if applicable the exact steps taken to verify that it works as expected._

### Notes (Optional)

_Specify here any other facts that you think are important for this issue._

### Attachments (Optional)

_Attachments like images, videos, etc. (drag and drop in the text box)_

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: src/script/components/calling/FullscreenVideoCall.tsx [hunk 1]
```diff
@@ -152,11 +151,11 @@ const FullscreenVideoCall: React.FC<FullscreenVideoCallProps> = ({
...
   const [hasUnreadMessages, setHasUnreadMessages] = useState(false);
-  const updateUnreadCount = (unreadCount: number) => setHasUnreadMessages(unreadCount > 0);
+  const {unreadMessagesCount} = useAppState();
+
   useEffect(() => {
-    amplify.subscribe(WebAppEvents.LIFECYCLE.UNREAD_COUNT, updateUnreadCount);
-    return () => amplify.unsubscribe(WebAppEvents.LIFECYCLE.UNREAD_COUNT, updateUnreadCount);
-  }, []);
+    setHasUnreadMessages(unreadMessagesCount > 0);
+  }, [unreadMessagesCount]);
```
Comment (PatrykBuniX) [inline src/script/components/calling/FullscreenVideoCall.tsx]: I'm not sure we need `hasUnreadMessages` to be separate local state (`useState` + `useEffect`). What about just creating a variable inside component `const hasUnreadMessages  = unreadMessagesCount > 0`, and always relaying on global zustand state (we will get rid of 1 extra re-render)? 😄

File: src/script/main/app.ts [hunk 2]
```diff
@@ -492,7 +492,6 @@ export class App {
       telemetry.timeStep(AppInitTimingsStep.APP_LOADED);
 
       telemetry.report();
-      amplify.publish(WebAppEvents.LIFECYCLE.LOADED);
```
Comment (PatrykBuniX) [inline src/script/main/app.ts]: are you sure about this change?

Comment (przemvs) [reply src/script/main/app.ts]: Yup, this one not needed :) This is not removed after migrate initialize app :)

File: src/script/components/calling/FullscreenVideoCall.tsx (hunk 1 above)
Comment (przemvs) [reply src/script/components/calling/FullscreenVideoCall.tsx]: Sure, good catch! :D

**Review by PatrykBuniX — APPROVED** [review]

File: src/script/page/useWindowTitle.ts [hunk 3]
```diff
@@ -52,88 +52,113 @@ export const useWindowTitle = () => {
...
 
-    const unreadConversations = unarchivedConversations.filter(conversationEntity => conversationEntity.hasUnread());
-    const unreadConversationsCount = unreadConversations.length;
-    const connectionRequestsCount = connectionRequests.length;
-
-    const unreadCount = connectionRequestsCount + unreadConversationsCount;
-
-    let specificTitle = unreadCount > MIN_UNREAD_COUNT ? `(${unreadCount}) ` : '';
-
-    amplify.publish(WebAppEvents.LIFECYCLE.UNREAD_COUNT, unreadCount);
```
Comment (atomrc) [inline src/script/page/useWindowTitle.ts:79]: ⚠️  This event still needs to be triggered as it's consumed by the desktop app

**Review by atomrc — CHANGES_REQUESTED** [review]

File: src/script/main/app.ts (hunk 2 above)
Comment (atomrc) [reply src/script/main/app.ts]: I think it's still needed for the desktop app. Please check

File: src/script/page/useWindowTitle.ts (hunk 3 above)
Comment (przemvs) [reply src/script/page/useWindowTitle.ts:79]: Reverted

File: src/script/main/app.ts (hunk 2 above)
Comment (przemvs) [reply src/script/main/app.ts]: Reverted :)

**Review by atomrc — APPROVED** [review]

**Pull request merged**